hazelcast-client-new/src/main/java/com/hazelcast/client/spi/impl/ClientSmartInvocationServiceImpl.java

```diff
@@ -45,10 +45,9 @@ public void invokeOnPartitionOwner(ClientInvocation invocation, int partitionId)
         if (owner == null) {
             throw new IOException("Partition does not have owner. partitionId : " + partitionId);
         }
-        ClientConnection connection =
-                (ClientConnection) connectionManager.getOrConnect(owner, authenticator);
         invocation.getClientMessage().setPartitionId(partitionId);
-        send(invocation, connection);
+        Connection connection = getConnection(owner);
```

Comment:
Wouldn't it be better if getConnection(owner) returns a client connection? Can a client have anything else than client connections?